### DISCUSSION TITLE
How do I get shared authentication between multiple Next.JS applications?

### DISCUSSION THREAD

**Discussion by dBianchii** [nextauthjs/next-auth#9136] [Help]
Hi everyone, I have a monorepo here: https://github.com/dBianchii/kodix-turbo

And I want to have multiple different Next.JS application, and all of them should point to `@kdx/kdx` (my main master backend) to authenticate. 

I have authentication working perfectly fine with Next-Auth V5 Beta. (I have a separate `@kdx/auth` package in there, which is being utilized to full extent by the main `@kdx/kdx` nextjs app).

I want to understand: how can I get authentication working on different NextJS environments with a shared authentication between them?

**Comment by leongieser** [comment]
For anyone coming across this in the future have a look at

https://github.com/typehero/typehero/tree/main/packages/auth/src

They used  "5.0.0-beta.4" to authenticate for the web app and the admin panel

**Reply by FaisalHafeez** [reply]
This is incredibly helpful, thank you! I was just trying to figure out a solution for something similar, and this resource is perfect. Thanks for pointing me in the right direction!
